### PR TITLE
context menu phase 2

### PR DESCRIPTION
**Pull request by MidfieId**
[for trello ticket](https://trello.com/c/z8KLyJMk/123-context-menu-phase-2)

### CODE REVIEW

**Comment by ahmed-okkio** [comment]
Make it so you cant right click yourself

File: Frontend/Frontend.Client/Components/ContextMenu.razor [hunk 1]
```diff
@@ -33,24 +39,32 @@
 
     public async void MuteUser()
     {
+        InfractionModalService.InfractionType = "mute";
```
Comment (ahmed-okkio) [inline Frontend/Frontend.Client/Components/ContextMenu.razor]: ContextMenuModalService

File: Frontend/Frontend.Client/Components/InfractionModal.razor [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+@using LSMP
+@page "/infraction"
+
+@if (InfractionModalService.modalShown)
+{
+    <div class="modal-overlay" @onclick="InfractionModalService.hideInfractionModal">
+        <div class="infraction-modal" @onclick:stopPropagation="true">
+            <div style="font-weight:bolder">@InfractionModalService.InfractionType.ToUpper() REASON</div>
+            <div class="reason-input-box">
+                <textarea placeholder="Type the reason here (leave empty incase of no reason)" value="@infractionReason" @oninput="e => infractionReason = e.Value.ToString()" />
```
Comment (ahmed-okkio) [inline Frontend/Frontend.Client/Components/InfractionModal.razor]: a simple "Reason..." would be fine

File: Frontend/Frontend.Client/Components/ContextMenu.razor (hunk 1 above)
Comment (MidfieId) [reply Frontend/Frontend.Client/Components/ContextMenu.razor]: changed names in all associated areas

File: Frontend/Frontend.Client/Components/InfractionModal.razor (hunk 2 above)
Comment (MidfieId) [reply Frontend/Frontend.Client/Components/InfractionModal.razor]: changed

File: Frontend/Frontend.Client/Components/ContextMenuModal.razor [hunk 3]
```diff
@@ -1,56 +1,55 @@
 ﻿﻿@using Models
 @using LSMP
-@page "/infraction"
```
Comment (ahmed-okkio) [inline Frontend/Frontend.Client/Components/ContextMenuModal.razor]: what zis

File: Frontend/Frontend.Client/Models/ContextMenuService.cs [hunk 4]
```diff
@@ -16,7 +17,7 @@ public static async Task ShowContextMenu(MouseEventArgs e, IJSRuntime jsRuntime,
         Top = e.ClientY;
         Left = e.ClientX;
         TargetUser = user;
-
+        if (ClientManager.CurrentUser == TargetUser) { return; }
```
Comment (ahmed-okkio) [inline Frontend/Frontend.Client/Models/ContextMenuService.cs]: no single single liners please

Comment (MidfieId) [reply Frontend/Frontend.Client/Models/ContextMenuService.cs]: made it a multi liner

**Review by ahmed-okkio — APPROVED** [review]

**Pull request merged**